odoo/addons/test_new_api/tests/test_properties.py

```diff
@@ -1294,6 +1295,214 @@ def test_properties_field_definition_update(self):
         self.assertEqual(expected_properties, sql_properties)
 
     @mute_logger('odoo.fields')
+    def test_properties_field_search(self):
+        self.env['test_new_api.message'].search(
+            [('id', 'not in', (self.message_1 | self.message_2 | self.message_3).ids)]
+        ).unlink()
+
+        # search on text properties
+        self.message_1.attributes = [{
+            'name': 'mychar',
+            'type': 'char',
+            'value': 'Test',
+            'definition_changed': True,
+        }]
+        self.message_2.attributes = {'mychar': 'TeSt'}
+
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', '=', 'Test')])
+        self.assertEqual(messages, self.message_1, "Should be able to search on a properties field")
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', '=', '"Test"')])
+        self.assertFalse(messages)
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', 'ilike', 'test')])
+        self.assertEqual(messages, self.message_1 | self.message_2)
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', 'not ilike', 'test')])
+        self.assertFalse(messages)
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', 'ilike', '"test"')])
+        self.assertFalse(messages)
+
+        for search in ['mychar!', 'my!char', '!mychar']:
+            with self.assertRaises(ValueError):
+                self.env['test_new_api.message'].search([(f'attributes.{search}', '=', 'Test')])
+
+        with self.assertRaises(AssertionError):
+            self.env['test_new_api.message'].search([('attributes.mychar.test', '=', 'Test')])
+
+        # search falsy properties
+        self.message_3.discussion = self.message_2.discussion
+        self.message_3.attributes = [{'name': 'mychar', 'value': False}]
+        self.assertEqual(self._get_sql_properties(self.message_3), {'mychar': False})
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', '=', False)])
+        self.assertEqual(messages, self.message_3)
+
+        # search falsy properties when the key doesn't exist in the dict
+        # message 2 properties is False, message 3 properties doesn't exist in database
+        self.message_2.attributes = [{'name': 'mychar', 'value': False}]
+        self.env.cr.execute(
+            "UPDATE test_new_api_message SET attributes = '{}' WHERE id = %s",
+            [self.message_3.id],
+        )
+        messages = self.env['test_new_api.message'].search([('attributes.mychar', '=', False)])
+        self.assertEqual(messages, self.message_2 | self.message_3)
```

Comment:
Doesn't that behavior seem a bit unintended though?

I can't imagine users searching `properties.is_completed = False` or `properties.sm_user_assigned = False` in project tasks and expecting to receive all tasks that don't have the "is_completed" or "sm_user_assigned" key defined on properties along with the ones that are specifically False.

Could we somehow make sure that the key exists and that it's Falsy? 
Or is it specifically not what we want for the feature? Seems very confusing to me.

(Maybe I missed something)